cpu/samd5x/periph/can.c

```diff
@@ -787,12 +812,13 @@ static void _isr(candev_t *candev)
         DEBUG_PUTS("protocol error in data phase");
         dev->conf->can->IR.reg |= CAN_IR_PED;
         /* Extract the Tx and Rx error counters */
-        uint8_t tx_err_cnt = (uint8_t)dev->conf->can->ECR.bit.TEC;
+        uint8_t tx_err_cnt = (uint8_t)(dev->conf->can->ECR.reg & CAN_ECR_TEC_Msk);
```

Comment:
```suggestion
        uint32_t ecr = dev->conf->can->ECR.reg
        uint8_t tx_err_cnt = (uint8_t)(ecr & CAN_ECR_TEC_Msk);
        uint8_t rx_err_cnt = (uint8_t)((ecr & CAN_ECR_REC_Msk) >> CAN_ECR_REC_Pos);
```
not sure about the uint32_t

Reply:
see other occurrence below

Reply:
Fixed.
`ECR` is a 32-bit wide register so `uint32_t` is fine here.